httpserver/https_client.cpp

```diff
@@ -31,20 +34,43 @@ void make_https_request(boost::asio::io_context& ioc,
     static ssl::context ctx{ssl::context::tlsv12_client};
 
     auto session = std::make_shared<HttpsClientSession>(
-        ioc, ctx, std::move(resolve_results), req, std::move(cb));
+        ioc, ctx, std::move(resolve_results), req, std::move(cb),
+        sn_pubkey_b32z);
 
     session->start();
 }
 
+static std::string x509_to_string(X509* x509) {
+    BIO* bio_out = BIO_new(BIO_s_mem());
+    PEM_write_bio_X509(bio_out, x509);
+    BUF_MEM* bio_buf;
+    BIO_get_mem_ptr(bio_out, &bio_buf);
+    std::string pem = std::string(bio_buf->data, bio_buf->length);
+    BIO_free(bio_out);
```

Comment:
Will return 0 on failure

Reply:
Not sure if we can do anything if the free operation failed...

Reply:
Except log it?

Reply:
print a message and `abort()` or something?